infra/src/test/kotlin/fi/fta/geoviite/infra/ratko/RatkoLocalServiceIT.kt

```diff
@@ -0,0 +1,287 @@
+
+        ratkoLocalService
+            .searchOperatingPoints(FreeText("BBB"))
+            .let { result -> assertEquals(3, result.size) }
+    }
+
+    @Test
+    fun `Operating points can be found by abbreviation`() {
+        val operatingPointAbbreviations = listOf(
+            "BBB",
+            "AAA",
+        )
+
+        operatingPointAbbreviations.map { abbreviation ->
+            createTestOperatingPoint(
+                name = "unused",
+                abbreviation = abbreviation,
+            )
+        }.let(operatingPointDao::updateOperatingPoints)
+
+        ratkoLocalService
+            .searchOperatingPoints(FreeText("BBB"))
+            .let { result -> assertEquals(1, result.size) }
+
+        ratkoLocalService
+            .searchOperatingPoints(FreeText("AAA"))
+            .let { result -> assertEquals(1, result.size) }
+    }
+
+
+    @Test
+    fun `Operating points search limit restricts the amount of results`() {
+        val stringThatMatchesAll = "Operating point"
+
+        (1..20).map { index ->
+            "$stringThatMatchesAll $index"
+        }.map { name ->
+            createTestOperatingPoint(
+                name = name,
+                abbreviation = "unused",
+            )
+        }.let(operatingPointDao::updateOperatingPoints)
+
+        ratkoLocalService
+            .searchOperatingPoints(FreeText(stringThatMatchesAll)) // Limit should be 10 by default.
+            .let { result -> assertEquals(10, result.size) }
+
+        ratkoLocalService
+            .searchOperatingPoints(FreeText(stringThatMatchesAll), resultLimit = 5)
+            .let { result -> assertEquals(5, result.size) }
+
+        ratkoLocalService
+            .searchOperatingPoints(FreeText(stringThatMatchesAll), resultLimit = 30)
+            .let { result -> assertEquals(20, result.size) }
+    }
+
+    @Test
+    fun `Operating points search results are sorted by name`() {
+        val operatingPointNames = listOf(
+            "BBB 1",
+            "CCC 1",
+            "AAA 1",
+        )
+
+        operatingPointNames.map { name ->
+            createTestOperatingPoint(
+                name = name,
+                abbreviation = "unused",
+            )
+        }.let(operatingPointDao::updateOperatingPoints)
+
+        ratkoLocalService
+            .searchOperatingPoints(FreeText("1"))
+            .let { result ->
+                assertEquals(3, result.size)
+
+                assertEquals(operatingPointNames[2], result[0].name)
+                assertEquals(operatingPointNames[0], result[1].name)
+                assertEquals(operatingPointNames[1], result[2].name)
+            }
+    }
+
+    @Test
+    fun `Operating points search only matches name or abbreviation, not both`() {
+        listOf(
+            createTestOperatingPoint(
+                name = "AAA",
+                abbreviation = "AAA",
+            )
+        ).let(operatingPointDao::updateOperatingPoints)
```

Comment:
Testin nimi lienee väärin, kun tässähän nimenomaan hakuehto matchaa molempiin?

Reply:
Ideanahan on siis tarkistaa ettei se mätsää kumpaakin erikseen 🤔. Jos se mätsäisi ne erikseen, niin tulisi kaksi kertaa sama hakutulos palautuslistassa (jos haku olisi esim. toteutettu kahdella sql-queryllä joista toinen hakisi nimellä ja toinen lyhenteellä, ja lopputulos olisi näiden yhdistetty lista ilman uniikkisuusfiltteriä, niin listassa olisi sama liikennepaikka kahdesti)

=> Vaihdettu `Operating points search matching both name and abbreviation returns the result only once`

Lieneeköhän sen kuvaavampi vain 😄